### PR TITLE
Added missing directional oriscus figures

### PR DESCRIPTION
**Pull request by henryso**
This includes full fusion support and covers the open items of #898 and its duplicate #972.

Note: I based this off of #1028, so merging this will merge that as well.  This is because I was forgot to change my branch when I started work.  If #0128 is to be rejected, I will rebase these changes.

Note 2: While I was coverage testing this change, I found a bug when passing a filename with no extension to gregorio, which I fixed.

I got somewhat confused during implementation, so any help testing with new and interesting combinations would be appreciated.  I have updated the tests to my satisfaction.

If this is found satisfactory, please merge it.

### CODE REVIEW

**Comment by henryso** [comment]
For those developing custom fonts, the following changes are necessary:

Glyph renames:
- `OriscusCavumHole` to `AscendensOriscusCavumHole`
- `OriscusCavum` to `AscendensOriscusCavum`
- `osbase` to `AscendensOriscusLineBLBR`
- `SalicusOriscus` to `AscendensOriscusLineBLTR`
- `OriscusLineBL` to `AscendensOriscusLineBL`
- `OriscusLineTR` to `AscendensOriscusLineTR`
- `Oriscus` to `AscendensOriscus`
- `OriscusCavumReversusHole` to `DescendensOriscusCavumHole`
- `OriscusCavumReversus` to `DescendensOriscusCavum`
- `OriscusReversusLineBL` to `DescendensOriscusLineBL`
- `odbase` to `DescendensOriscusLineBR`
- `OriscusReversusLineTLBR` to `DescendensOriscusLineTLBR`
- `OriscusReversusLineTL` to `DescendensOriscusLineTL`
- `OriscusReversus` to `DescendensOriscus`

New glyphs (should be self-explanatory):
- `AscendensOriscusLineBR`
- `AscendensOriscusLineTLBR`
- `AscendensOriscusLineTLTR`
- `AscendensOriscusLineTL`
- `DescendensOriscusLineBLBR`
- `DescendensOriscusLineBLTR`
- `DescendensOriscusLineTLTR`
- `DescendensOriscusLineTR`

File: UPGRADE.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ High notes are now taken into account in vertical spacings. If you would like to
 
 `\grelooseness` is now `-1` by default. This should prevent orphaned syllable at end of score. If you prefer the old behavior, use `\def\grelooseness{\looseness}` befor including your scores.
 
+### Stemmed oriscus flexus orientation
```
Comment (eroux) [inline UPGRADE.md]: I think something is wrong with this file, it should have a `4.1.1` entry...

Comment (henryso) [inline UPGRADE.md]: This is a problem in 4.1.1 itself.  I'll fix it for develop since it's too late for 4.1.1.

**Comment by eroux** [comment]
Thanks! merging

**Pull request merged**